PES-1377 Add carrier settings for country page

Mrakor opened the pull request.

Mrakor committed:
PES-1377 Add carrier settings for country page

fixup! PES-1377 Add carrier settings for country page

fixup! PES-1377 Add carrier settings for country page

Commit:
PES-1377 Remove column for carrier cart name

Commit:
PES-1377 Remove yet unused translation

packeta-user commented on upload/admin/view/template/extension/shipping/zasilkovna_carrier_settings.twig:
```diff
@@ -23,7 +28,11 @@
                                     <thead>
                                     {% for column_title in columns_titles %}
-                                        <td class="text-left">
+                                        {% if loop.last %}
```
Záhlaví děláme v drtivé většině klasickým způsobem a ne iterací pole.  Je to přehlednější.  Zůstal bych u toho i v této šabloně.

packeta-user commented on upload/admin/view/template/extension/shipping/zasilkovna_carrier_settings_country.twig:
```diff
@@ -0,0 +1,66 @@
+                            <div class="table-responsive">
+                                <table class="table table-bordered table-hover">
+                                    <thead>
+                                    {% for column_title in columns_titles %}
```
Totéž co v předchozím komentáři.  Záhlaví děláme v drtivé většině klasickým způsobem a ne iterací pole. Je to přehlednější. Zůstal bych u toho i v této šabloně.

Mrakor committed:
PES-1377 Add flashMessage for countries where Packeta doesnt deliver

Fix display of carrier names if overriden by name_cart

Mrakor committed:
PES-1377 CR fix - explicit column titles in template

Mrakor committed:
PES-1377 Use explicit column names in template

packeta-user commented on upload/admin/language/cs-cz/extension/shipping/zasilkovna.php:
```diff
@@ -168,6 +168,14 @@
+$_['carrier_settings_country_column_name'] = 'Název dopravce';
+$_['carrier_settings_country_column_action'] = 'Akce';
+$_['carrier_settings_country_not_found'] = 'Země s kódem "%s" nebyla nalezena.';
+$_['carrier_settings_packeta_doesnt_deliver_to_country'] = '%s: Zásilkovna do této země nedoručuje.';
```
Připadá mi lepší hláška:  Zásilkovna do země "%s" nedoručuje.

packeta-user commented on upload/admin/controller/extension/shipping/zasilkovna.php:
```diff
@@ -1064,19 +1071,72 @@ public function carrier_settings()
+		$this->load->model('extension/shipping/zasilkovna_countries');
+
+		$countryCode = $this->request->get['country'];
+		$countryName = $this->model_extension_shipping_zasilkovna_countries->getCountryNameByIsoCode2($countryCode);
```
Není lepší použít v tomto případě getCountryByIsoCode2() ? Podmínka zda země existuje bude přehlednější.

packeta-user commented on upload/admin/controller/extension/shipping/zasilkovna.php:
```diff
@@ -1064,19 +1071,72 @@ public function carrier_settings()
+		$countryListingPage = $this->diContainer->get(\Packetery\Carrier\CountryListingPage::class);
+		$activeCountries = $countryListingPage->getActiveCountries();
+
+		if (!in_array($countryCode, array_column($activeCountries, 'code'), true)) {
```
Tady by se spíše hodila metoda, které předáš jako parametr kód země a vrátí se informace, zda Zásilkovna do této země doručuje.   Metoda bude tedy vracet:  true/false

packeta-user commented on upload/admin/controller/extension/shipping/zasilkovna.php:
```diff
@@ -1064,19 +1071,72 @@ public function carrier_settings()
+		$vendors = $this->vendorRepository->getVendorsByCountry($countryCode);
+		if ($vendors) {
+			array_walk($vendors, static function (&$vendor) {
+				$vendor['display_name'] = $vendor['carrier_name_cart'] ?: $vendor['name'];
```
Logika zda se má použít name nebo name_cart by měla být v šabloně.

packeta-user commented on upload/system/library/Packetery/Vendor/VendorRepository.php:
```diff
@@ -0,0 +1,52 @@
+	/** @var DB */
+	private $db;
+
+	public function __construct(DB $db) {
```
chybí PHP DOC

Mrakor committed:
PES-1377 CR fixes

Add PHP annotation for VendorRepository constructor
Introduce doesPacketaDeliverTo
move display name logic to template
use getCountryByIsoCode2 instead of getCountryNameByIsoCode2
improve warning flashMessage
